### PR TITLE
display coin flip error better CORE-10306

### PR DESCRIPTION
**Pull request by mmaxim**
Patch does the following:

1.) Sends up error information to the UI so it can construct custom displays depending on the type of error.
2.) Create a custom display for absentee errors, the most common kind.

cc @adamjspooner  @malgorithms

### CODE REVIEW

**Comment by chrisnojima** [comment]
is errors missing?

**Comment by mmaxim** [comment]
Sigh, sorry

File: shared/chat/conversation/messages/coinflip/errors.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  error: RPCChatTypes.UICoinFlipError,
+|}
+
+const CoinFlipError = (props: Props) => {
+  if (props.error.typ === RPCChatTypes.chatUiUICoinFlipErrorTyp.generic && props.error.generic) {
+    return <CoinFlipGenericError error={props.error.generic} />
+  } else if (props.error.typ === RPCChatTypes.chatUiUICoinFlipErrorTyp.absentee && props.error.absentee) {
+    return <CoinFlipAbsenteeError error={props.error.absentee} />
+  }
+  return <CoinFlipGenericError error={'Unknown error occurred'} />
```
Comment (buoyad) [inline shared/chat/conversation/messages/coinflip/errors.js]: `error="..."`

File: shared/chat/conversation/messages/coinflip/errors.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import * as Kb from '../../../../common-adapters'
+import * as Styles from '../../../../styles'
+import * as RPCChatTypes from '../../../../constants/types/rpc-chat-gen'
+
+type Props = {|
+  error: RPCChatTypes.UICoinFlipError,
+|}
+
+const CoinFlipError = (props: Props) => {
+  if (props.error.typ === RPCChatTypes.chatUiUICoinFlipErrorTyp.generic && props.error.generic) {
```
Comment (buoyad) [inline shared/chat/conversation/messages/coinflip/errors.js]: :sheep: `switch` might be better

**Pull request merged**